api/tests/test_canteens.py

```diff
@@ -663,23 +697,119 @@ def test_get_canteen_actions(self):
+
+        canteen.satellite_canteens_count = 123
+        canteen.save()
+
+        response = self.client.get(reverse("list_actionable_canteens", kwargs={"year": 2021}))
+        returned_canteens = response.json()["results"]
+        self.assertEqual(returned_canteens[0]["action"], "40_teledeclare")
+
+        # Satellites should have the SIRET of the central cuisine
+        canteen.production_type = Canteen.ProductionType.ON_SITE_CENTRAL
+        canteen.central_producer_siret = None
+        canteen.save()
+
+        response = self.client.get(reverse("list_actionable_canteens", kwargs={"year": 2021}))
+        returned_canteens = response.json()["results"]
+        self.assertEqual(returned_canteens[0]["action"], "35_fill_canteen_data")
+
+        canteen.central_producer_siret = 75665621899905
+        canteen.save()
+
+        response = self.client.get(reverse("list_actionable_canteens", kwargs={"year": 2021}))
+        returned_canteens = response.json()["results"]
+        self.assertEqual(returned_canteens[0]["action"], "40_teledeclare")
+
     @override_settings(ENABLE_TELEDECLARATION=True)
     @authenticate
     def test_get_diagnostics_to_td(self):
         """
         Check that the actions endpoint includes a list of diagnostics that could be teledeclared
         """
         last_year = 2021
-        no_diag = CanteenFactory.create()
-        canteen_with_incomplete_diag = CanteenFactory.create()
+        no_diag = CanteenFactory.create(
+            production_type=Canteen.ProductionType.ON_SITE,
+            publication_status=Canteen.PublicationStatus.PUBLISHED,
+            management_type=Canteen.ManagementType.DIRECT,
+            yearly_meal_count=1000,
+            daily_meal_count=12,
+            siret="75665621899905",
+            city_insee_code="69123",
+        )
+        canteen_with_incomplete_diag = CanteenFactory.create(
+            production_type=Canteen.ProductionType.ON_SITE,
+            publication_status=Canteen.PublicationStatus.PUBLISHED,
+            management_type=Canteen.ManagementType.DIRECT,
+            yearly_meal_count=1000,
+            daily_meal_count=12,
+            siret="96766910375238",
+            city_insee_code="69123",
+        )
         DiagnosticFactory.create(canteen=canteen_with_incomplete_diag, year=last_year, value_total_ht=None)
-        canteen_with_complete_diag = CanteenFactory.create()
+        canteen_with_complete_diag = CanteenFactory.create(
+            production_type=Canteen.ProductionType.ON_SITE,
+            publication_status=Canteen.PublicationStatus.PUBLISHED,
+            management_type=Canteen.ManagementType.DIRECT,
+            yearly_meal_count=1000,
+            daily_meal_count=12,
+            siret="75665621899905",
+            city_insee_code="69123",
+        )
         complete_diag = DiagnosticFactory.create(
             canteen=canteen_with_complete_diag, year=last_year, value_total_ht=10000
         )
```

Comment:
Adding an `incomplete_canteen_complete_diag` would test to see if the mass TD ignores this case

Reply:
Added here : https://github.com/betagouv/ma-cantine/pull/2312/commits/1ef72d22b031e83e0fd3d431f6b086139cc1c298